samcli/commands/_utils/template.py

```diff
@@ -29,20 +29,26 @@ class TemplateFailedParsingException(UserException):
     pass
 
 
-def get_template_data(template_file):
+def get_template_data(template_file, root_template_path: str = ""):
     """
     Read the template file, parse it as JSON/YAML and return the template as a dictionary.
 
     Parameters
     ----------
     template_file : string
         Path to the template to read
+    root_template_path: string
+        Path of the root SAM Template
 
     Returns
     -------
     Template data as a dictionary
     """
 
+    # if template_file is a relative path , we use root_template_path to convert to absolute path
+    if not os.path.isabs(template_file) and root_template_path:
+        template_file = os.path.join(root_template_path, template_file)
```

Comment:
It looks to me that we need to get dirname from `root_template_path` before joining

Reply:
Actually the name is misleading. I just updated it to `root_template_dir`